mreg_cli/api/models.py

```diff
@@ -3157,6 +3194,70 @@ def get_by_any_means(
 
         return host
 
+    @classmethod
+    def get_list_by_any_means(
+        cls, identifier: str, inform_as_cname: bool = True, inform_as_ptr: bool = True
+    ) -> list[Self]:
+        """Get a host by the given identifier.
+
+        - If the identifier is numeric, it will be treated as an ID.
+        - If the identifier is an IP address, it will be treated as an IP address (v4 or v6).
+        - If the identifier is a MAC address, it will be treated as a MAC address.
+        - Otherwise, it will be treated as a hostname. If the hostname is a CNAME,
+        the host it points to will be returned.
+
+        To check if a returned host is a cname, one can do the following:
+
+        ```python
+        hostname = "host.example.com"
+        host = get_host(hostname, ok404=True)
+        if host is None:
+            print("Host not found.")
+        elif host.name != hostname:
+            print(f"{hostname} is a CNAME pointing to {host.name}")
+        else:
+            print(f"{host.name} is a host.")
+        ```
+
+        Note that get_host will perform a case-insensitive search for a fully qualified version
+        of the hostname, so the comparison above may fail.
+
+        :param identifier: The identifier to search for.
+        :param ok404: If True, don't raise a EntityNotFound if the host is not found.
+        :param inform_as_cname: If True, inform the user if the host is a CNAME.
+        :param inform_as_ptr: If True, inform the user if the host is a PTR override.
+
+        :raises EntityNotFound: If we don't find the host and `ok404` is False.
+
+        :returns: A Host object if the host was found, otherwise None.
+        """
+        if identifier.isdigit():
+            return cls.get_list_by_id(int(identifier))
+
+        if ip := NetworkOrIP.parse(identifier, mode="ip"):
+            return cls.get_list_by_ip(ip, inform_as_ptr=inform_as_ptr)
+
+        if mac := MacAddress.parse(identifier):
+            return cls.get_list_by_mac_or_raise(mac)
+
+        # Let us try to find the host by name...
+        identifier = HostName.parse_or_raise(identifier)
+
+        if hosts := cls.get_list_by_field("name", identifier):
+            return hosts
```

Comment:
Similarly here and with CNAMES, wherein a name or ID should point to a _single_ host, not multiple, surely?

Reply:
Yeah, hostnames can't be shared. Neither can CNAMEs. They can have multiple IPs and such, but they're unique and distinct host/cname objects.